Sources/Common/Background Queue/QueueStorage.swift

```diff
@@ -147,42 +147,49 @@ public class FileManagerQueueStorage: QueueStorage {
 
         logger.debug("deleting expired tasks from the queue")
 
-        var tasksToDelete: Set<QueueTaskMetadata> = Set()
-        let queueTaskExpiredThreshold = Date().subtract(sdkConfig.backgroundQueueExpiredSeconds, .second)
-        logger.debug("""
-        deleting tasks older then \(queueTaskExpiredThreshold.string(format: .iso8601noMilliseconds)),
-        current time is: \(Date().string(format: .iso8601noMilliseconds))
-        """)
-
-        getInventory().filter { inventoryItem in
-            // Do not delete tasks that are at the start of a group of tasks.
-            // Why? Take for example Identifying a profile. If we identify profile X in an app today,
-            // we expire the Identify queue task and delete the queue task, and then profile X stays logged
-            // into an app for 6 months, that means we run the risk of 6 months of data never successfully being sent
-            // to the API.
-            // Also, queue tasks such as Identifying a profile are more rare queue tasks compared to tracking of events
-            // (that are not the start of a group). So, it should rarely be a scenario when there are thousands
-            // of "expired" Identifying a profile tasks sitting in a queue. It's the queue tasks such as tracking
-            // that are taking up a large majority of the queue inventory. Those we should be deleting more of.
-            inventoryItem.groupStart == nil
-        }.forEach { taskInventoryItem in
-            let isItemTooOld = taskInventoryItem.createdAt.isOlderThan(queueTaskExpiredThreshold)
-
-            if isItemTooOld {
-                tasksToDelete.insert(taskInventoryItem)
-            }
-        }
-
-        logger.debug("deleting \(tasksToDelete.count) tasks. \n Tasks: \(tasksToDelete)")
-
-        tasksToDelete.forEach { taskToDelete in
-            // Because the queue tasks we are deleting are not the start of a group,
-            // if deleting a task is not successful, we can ignore that
-            // because it doesn't negatively effect the state of the SDK or the queue.
-            _ = self.delete(storageId: taskToDelete.taskPersistedId)
-        }
-
-        return Array(tasksToDelete)
+        // When SdkConfig was modified to remove unused config options, this function could no longer run
+        // since the expired config option was removed.
+
+        // Commenting this code out for now, in case it's decided we still need it. Otherwise, this function can be deleted.
```

Comment:
If the code is not required by any class for running the migrations, I would vote for removing it completely.

Reply:
Done.